### DISCUSSION TITLE
Application repositories connection

### DISCUSSION THREAD

**Discussion by cnukwas** [argoproj-labs/applicationset#295] [Q&A] [answered]
Hi,

Our internal github requires AD/LDAP users login, even to view any repositories, so wondering how application teams should set up Argo repository connections for each Application that's being deployed to Kubernetes/OpenShift platform.
Users are able to view and clone any repositories once he/she is logged in, but not otherwise.
Would like to avoid individual user credentials input for setting up a repo connection since Kubernetes secrets are not encrypted(cluster and Argo admin is controlled, we still don't users credentials stored by Argo Configmap or Secrets).

Are there any workarounds or solutions to avoid either credentials or SSH keys input for each team and application repository, but still allow the teams to use a repo for Application configuration?

Thanks

**Comment by jgwest** [comment, marked as answer]
You may want to ask on https://github.com/argoproj/argo-cd/discussions , as it sounds like this is an Argo CD specific queston, but in general Argo CD will need to have either credentials or an SSH key for any of the repositories that it connects to, as per [this](https://argoproj.github.io/argo-cd/user-guide/private-repositories/), I'm not aware of a way around this.

**Reply by cnukwas** [reply]
Sorry about this, I thought I was in the Argo CD github. @jgwest, thanks for the information. I see something about [credentials templates](https://argoproj.github.io/argo-cd/user-guide/private-repositories/#credential-templates) in the referenced documentation, not sure if that's something we can set up, if the id has at least 'read only' access to all github enterprise repositories. I will also post in Argo CD forum and go from there.
